api/app.ts

```diff
@@ -1121,6 +1121,10 @@ app.post(
 				)
 			}
 			let settings = req.body
+			// Extract restart flag from request body, default to true for backward compatibility
+			const shouldRestart = settings.restart !== false
```

Comment:
instead of passing restart in settings and delete it change body to be { restart, data }